### ISSUE TITLE
Bug: JsonWidget dont appears

### ISSUE THREAD

**Issue by glanzel** [cursive-works/wagtail-json-widget#11]
I used your demo code + adding a JsonBlock to it:

```
from django import forms
from wagtail.fields import StreamField 
from wagtail.admin.edit_handlers import FieldPanel
from wagtail.admin.forms import WagtailAdminPageForm
from wagtail.core.models import Page
from wagtail_json_widget.widgets import JSONEditorWidget
from django.db import models
from wagtail.models import Page
from wagtail_json_widget import blocks as jblocks

class HomePage(Page):
    pass

class MyPageForm(WagtailAdminPageForm):
    myjson = forms.JSONField(widget=JSONEditorWidget, required=False)


class MyPage(Page):

    myjson = models.JSONField(null=True, blank=True)
    body = StreamField([
        ("features", jblocks.JSONBlock()), 
        ("conditions", jblocks.JSONBlock())
        ], use_json_field=True, null=True)


    content_panels = Page.content_panels + [
        FieldPanel('myjson'), FieldPanel('body'),
    ]
    base_form_class = MyPageForm

```
but the myjson Field stays empty without complaining while the JsonBlock tells me "This widget failed to render, please check the console for details".
The JS Console says:
`Uncaught TypeError: this.widget is undefined`
and
`SyntaxError: JSON.parse: unexpected end of data at line 1 column 1 of the JSON data`

I added the app to INSTALLED_APPS.

**Comment by marts** [comment]
Hi @glanzel thanks for the report. I've just tested on Wagtail 4.2 and get the same problem, so it may be something has changed in a recent Wagtail release.

What version of Wagtail are you using?

**Comment by marts** [comment]
@glanzel it looks like there is a problem parsing an empty string, as that is not valud JSON. I've worked around the issue by setting a default value, e.g.

```
body = StreamField([
        ("features", jblocks.JSONBlock(default="null")), 
        ("conditions", jblocks.JSONBlock(default="{}"))
        ], use_json_field=True, null=True)
```

https://stackoverflow.com/questions/18419428/what-is-the-minimum-valid-json

Does this work for you?

**Comment by glanzel** [comment]
Thanks, Tht helped me. It now works for the Streanḿfield/JSONBlock.  (fyi:
It does still not work for the JsonField but thats not important  for me )


Am Di., 14. Feb. 2023 um 18:23 Uhr schrieb Martin Swarbrick <
***@***.***>:

**Comment by marts** [comment]
@glanzel great that it works for your streamfield blocks...I'll look into the JsonField problem and try to get a more permanent fix released.

**Comment by marts** [comment]
Looks like Wagtail 4 changed some of the generated class names in the admin interface by adding the `w-field` prefix.

Probably need to update the class names, e.g. `w-field--json_editor_widget` and `w-field__input` or find a different way of doing it that doesn't rely on Wagtail generated classes.

https://github.com/cursive-works/wagtail-json-widget/blob/fb980f5d88c9383bface6786569a61bbe1437327/wagtail_json_widget/static/wagtail_json_widget/js/json_widget.js#L25

**Comment by marts** [comment]
@glanzel I've released a new version (0.0.7) to pypi that should now work with Wagtail 4
